microsetta_public_api/api/taxonomy.py

```diff
@@ -162,3 +161,43 @@ def _exists(resource, samples, taxonomy_repo):
         return missing_resource
 
     return jsonify(taxonomy_repo.exists(samples, resource)), 200
+
+
+def ranks_sample(dataset, resource, sample_size):
+    taxonomy_repo = _get_taxonomy_repo(dataset)
+
+    error_response = _check_resource_and_missing_ids(taxonomy_repo,
```

Comment:
So this isn't in code that has been changed in this PR, but if we change the calls to `check_missing_ids` and `validate_resource` in the the `_check_resource_and_missing_ids` function to `check_missing_ids_alt` and `validate_resource_alt`, we can avoid the

```python
error = _check_something()
if error:
    return error
```

if favor of, simply:
```python
_check_something()
```

But if that involves changing too much out-of-scope code we can open a new issue.

Reply:
Sounds good, punted to an issue